### PR TITLE
Support s390x BE architecture

### PR DESCRIPTION
**Pull request by shahidhs-ibm**
This PR adds support for IBM Z 64-bit BE (s390x) Linux.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -29,7 +29,11 @@ check-go-version:
 ESBUILD_RACE ?= -race
 
 test-go:
+ifeq (s390x,$(shell uname -p))
+	go test ./internal/...
+else
```
Comment (evanw) [inline Makefile:34]: Why is this necessary?

Comment (shahidhs-ibm) [reply Makefile:34]: `-race` option is not supported on s390x arch. If we don't remove this flag for s390x, we get following error:
```
go test: -race is only supported on linux/amd64, linux/ppc64le, linux/arm64, freebsd/amd64, netbsd/amd64, darwin/amd64, darwin/arm64, and windows/amd64
```

Comment (evanw) [reply Makefile:34]: Ok, good to know. I'll disable it by default in the future then.

File: lib/npm/node-platform.ts [hunk 2]
```diff
@@ -24,6 +24,7 @@ export const knownUnixlikePackages: Record<string, string> = {
   'linux ia32 LE': 'esbuild-linux-32',
   'linux mips64el LE': 'esbuild-linux-mips64le',
   'linux ppc64 LE': 'esbuild-linux-ppc64le',
+  'linux s390x BE': 'esbuild-linux-s390x',
```
Comment (evanw) [inline lib/npm/node-platform.ts:27]: I just realized that this is our first big-endian platform! I think everything should work fine but we'll see. If something doesn't work about the published package or install script, we can fix it later.

**Review by evanw — APPROVED** [review]
This looks good. If all goes well, this should be supported starting with the next published version of esbuild whenever that goes out.
